src/ray/core_worker/core_worker.cc

```diff
@@ -505,6 +502,15 @@ void CoreWorker::RunIOService() {
   io_service_.run();
 }
 
+void CoreWorker::WaitForShutdown() {
+  if (io_thread_.joinable()) {
```

Comment:
Do we need this joinable check?  We only join this once, right?

Reply:
Yes, we only join it once. But since this method has no sense about when it'll be called, I want to make it more robust by allowing it to be called multiple times.